src/utils.ts

```diff
@@ -0,0 +1,111 @@
+import { defaultAbiCoder } from "@ethersproject/abi";
+import { impersonateAccount, setBalance } from "@nomicfoundation/hardhat-network-helpers";
+import { NumberLike } from "@nomicfoundation/hardhat-network-helpers/dist/src/types";
+import { expect } from "chai";
+import { ContractInterface, TransactionResponse } from "ethers";
+import { ethers, network } from "hardhat";
+
+import { Command, Proposal, ProposalMeta, ProposalType } from "./types";
+import VENUS_CHAINLINK_ORACLE_ABI from "./vip-framework/abi/VenusChainlinkOracle.json";
+import COMPTROLLER_ABI from "./vip-framework/abi/comptroller.json";
+
+const COMPTROLLER = "0xfd36e2c2a6789db23113685031d7f16329158384";
+
+export async function setForkBlock(blockNumber: number) {
+  await network.provider.request({
+    method: "hardhat_reset",
+    params: [
+      {
+        forking: {
+          jsonRpcUrl: process.env.BSC_ARCHIVE_NODE,
+          blockNumber: blockNumber,
+        },
+      },
+    ],
+  });
+}
+
+export function getCalldatas({ signatures, params }: { signatures: string[]; params: any[][] }) {
+  return params.map((args: any[], i: number) => {
+    const types = getArgs(signatures[i]);
+    return defaultAbiCoder.encode(types, args);
+  });
+}
+
+const getArgs = (func: string) => {
+  if (func === "") return [];
+  // First match everything inside the function argument parens.
+  const match = func.match(/.*?\(([^)]*)\)/);
+  const args = match ? match[1] : "";
+  // Split the arguments string into an array comma delimited.
+  return args
+    .split(",")
+    .map(arg => {
+      // Ensure no inline comments are parsed and trim the whitespace.
+      return arg.replace(/\/\*.*\*\//, "").trim();
+    })
+    .filter(arg => {
+      // Ensure no undefined values are added.
+      return arg;
+    });
+};
+
+export const initMainnetUser = async (user: string, balance: NumberLike) => {
+  await impersonateAccount(user);
+  await setBalance(user, balance);
+  return ethers.getSigner(user);
+};
+
+export const makeProposal = (commands: Command[], meta: ProposalMeta, type: ProposalType): Proposal => {
+  return {
+    signatures: commands.map(cmd => cmd.signature),
+    targets: commands.map(cmd => cmd.target),
+    params: commands.map(cmd => cmd.params),
+    values: commands.map(cmd => cmd.value ?? "0"),
+    meta,
+    type,
+  };
+};
+
+export const setMaxStalePeriodInOracle = async (
+  comptrollerAddress: string,
+  maxStalePeriodInSeconds: number = 31536000 /* 1 year */,
+) => {
+  const provider = ethers.provider;
+  const comptroller = new ethers.Contract(COMPTROLLER, COMPTROLLER_ABI, provider);
```

Comment:
I would use the `comptrollerAddress` parameter, instead of the constant `COMPTROLLER`

Moreover, I wouldn't add the abi's of Comptroller and VenusChainlinkOracle. I would we don't need them to interact with the contract in this case, and we would reduce dependencies

Reply:
I think we would require abi's to create instance of comptroller to get the address of oracle and oracle instance to get the admin of oracle.

Reply:
You are right